src/components/biggive-tipping-slider/biggive-tipping-slider.tsx

```diff
@@ -0,0 +1,117 @@
+import { Component, Element, Prop, h } from '@stencil/core';
+import { spacingOption } from '../../globals/spacing-option';
+import { brandColour } from '../../globals/brand-colour';
+
+@Component({
+  tag: 'biggive-tipping-slider',
+  styleUrl: 'biggive-tipping-slider.scss',
+  shadow: true,
+})
+export class BiggiveTippingSlider {
+  @Element() host: HTMLBiggiveTippingSliderElement;
+
+  @Prop() spaceBelow: spacingOption = 0;
+
+  @Prop() colourScheme: brandColour = 'primary';
+
+  @Prop() percentageStart: number;
+
+  @Prop() percentageEnd: number;
+
+  @Prop() donationAmount: number;
+
+  @Prop() donationCurrency: string = '£';
+
+  componentDidRender() {
+    var isMoving = false;
+    var handle = this.host.shadowRoot?.querySelector<HTMLElement>('.handle');
+    var bar = this.host.shadowRoot?.querySelector<HTMLElement>('.bar');
+    var percentageWrap = handle?.querySelector('.percentage-value');
+    var donationWrap = handle?.querySelector('.donation-value');
+
+    var move = e => {
```

Comment:
`e` is implicitly any. PhpStorm suggests changing it to ` MouseEvent | TouchEvent`, I don't know if that's the best type for it.